multipart/multipart.py

```diff
@@ -696,8 +697,7 @@ class OctetStreamParser(BaseParser):
     """
 
     def __init__(self, callbacks: OctetStreamCallbacks = {}, max_size: float = float("inf")):
-        super().__init__()
-        self.callbacks = callbacks
+        super().__init__(callbacks)
```

Comment:
The problem here is that those are not type check compliant.

Reply:
Hmm, I am not sure I understand what you mean. I do get a warning if, for example, I replace callbacks with a `1`.

In my IDE:
```
Diagnostics:
1. Argument of type "Literal[2]" cannot be assigned to parameter "callbacks" of type "BaseCallbacks" in function "__init__"
     "Literal[2]" is incompatible with "BaseCallbacks" [reportArgumentType]
```

Or running mypy:
```
multipart/multipart.py:700: error: Argument 1 to "__init__" of "BaseParser" has incompatible type "int"; expected "BaseCallbacks"  [arg-type]
```

Reply:
I meant on the inherited classes. If I call the `super().__init__(callbacks)`, should error. Right?

Reply:
It is working for me. Would definitely be better with a named arg.


Also the tests are passing, and they do initialize the different parsers.

Reply:
would you prefer `super().__init__(callbacks=callbacks)`? Or none at all?

Reply:
I'm unsure we need a base class here...

Reply:
A Base class already exists, and expects both in it's description and it's methods (i.e `self.callback`, that an attribute `self.callbacks` exists).

That is why I see an error in line 595 (as of commit 3b15e38):

<img width="630" alt="image" src="https://github.com/Kludex/python-multipart/assets/33374178/bc1a2f78-3743-4a34-941a-a24094308e48">

In my opinion, this BaseParser class should:
1. Be an abstract class
2. Avoid initialization like this:
<img width="438" alt="image" src="https://github.com/Kludex/python-multipart/assets/33374178/6a1a4d73-2c73-4135-8583-a021ab733d6f">
It feels weird to me for an object to rely on an attribute that is not defined at all. 
The method `self.callback` will fail if the dictionary `self.callbacks` is not added after initialization.

Reply:
Can you create a PR proposing those changes?